src/Advection/centered_reconstruction.jl

```diff
@@ -0,0 +1,116 @@
+#####
+##### Centered advection scheme
+#####
+
+"""
+    struct Centered <: AbstractCenteredAdvectionScheme{2}
+
+Centered reconstruction scheme.
+"""
+struct Centered{N, FT, XT, YT, ZT, CA} <: AbstractCenteredAdvectionScheme{N, FT} 
+    "coefficient for Centered reconstruction on stretched x-faces" 
+    coeff_xᶠᵃᵃ::XT
+    "coefficient for Centered reconstruction on stretched x-centers"
+    coeff_xᶜᵃᵃ::XT
+    "coefficient for Centered reconstruction on stretched y-faces"
+    coeff_yᵃᶠᵃ::YT
+    "coefficient for Centered reconstruction on stretched y-centers"
+    coeff_yᵃᶜᵃ::YT
+    "coefficient for Centered reconstruction on stretched z-faces"
+    coeff_zᵃᵃᶠ::ZT
+    "coefficient for Centered reconstruction on stretched z-centers"
+    coeff_zᵃᵃᶜ::ZT
+
+    "advection scheme used near boundaries"
+    boundary_scheme :: CA
```

Comment:
Or perhaps `buffer_scheme`, which is invoked within `buffer` points of a boundary